src/libraries/System.Net.Http/src/System/Net/Http/Headers/AltSvcHeaderParser.cs

```diff
@@ -0,0 +1,488 @@
+            {
+                if (idx != value.Length)
+                {
+                    // Clear has no parameters and should be the only Alt-Svc value present, so there should be nothing after it.
+                    parsedValue = null;
+                    return 0;
+                }
+
+                parsedValue = AltSvcHeaderValue.Clear;
+                return idx - startIndex;
+            }
+
+            if (idx == value.Length || value[idx++] != '=')
+            {
+                parsedValue = null;
+                return 0;
+            }
+
+            if (!TryReadQuotedAltAuthority(value, idx, out string altAuthorityHost, out int altAuthorityPort, out int altAuthorityLength))
+            {
+                parsedValue = null;
+                return 0;
+            }
+            idx += altAuthorityLength;
+
+            // Parse parameters: *( OWS ";" OWS parameter )
+            int? maxAge = null;
+            bool persist = false;
+
+            while (idx != value.Length)
+            {
+                // Skip OWS before semicolon.
+                while (idx != value.Length && IsOptionalWhiteSpace(value[idx])) ++idx;
+
+                if (idx == value.Length || value[idx++] != ';')
+                {
+                    parsedValue = null;
+                    return 0;
+                }
+
+                // Skip OWS after semicolon / before value.
+                while (idx != value.Length && IsOptionalWhiteSpace(value[idx])) ++idx;
+
+                // Get the parameter key length.
+                int tokenLength = HttpRuleParser.GetTokenLength(value, idx);
+                if (tokenLength == 0)
+                {
+                    parsedValue = null;
+                    return 0;
+                }
+
+                if ((idx + tokenLength) >= value.Length || value[idx + tokenLength] != '=')
+                {
+                    parsedValue = null;
+                    return 0;
+                }
+
+                if (tokenLength == 2 && value[idx] == 'm' && value[idx + 1] == 'a')
+                {
+                    // Parse "ma" (Max Age).
+
+                    idx += 3; // Skip "ma="
+                    if (!TryReadTokenOrQuotedInt32(value, idx, out int maxAgeTmp, out int parameterLength))
+                    {
+                        parsedValue = null;
+                        return 0;
+                    }
+
+                    if (maxAge == null)
+                    {
+                        maxAge = maxAgeTmp;
+                    }
+                    else
+                    {
+                        // RFC makes it unclear what to do if a duplicate parameter is found. For now, take the minimum.
+                        maxAge = Math.Min(maxAge.GetValueOrDefault(), maxAgeTmp);
+                    }
+
+                    idx += parameterLength;
+                }
+                else if (value.AsSpan(idx).SequenceEqual("persist="))
```

Comment:
Is this correct?  If something comes after the persist=, like a value, that will be included in the span, in which case even if it starts with persist= it won't equal it.  Did you mean to use StartsWith here?